js/options.js

```diff
@@ -49,7 +49,68 @@ document.addEventListener('DOMContentLoaded', function () {
     });
   }
 
+  // Helper for export function
+  let textFile = null;
+  function makeTextFile(text) {
+    var data = new Blob([text], { type: 'text/plain' });
+    // If we are replacing a previously generated file we need to
+    // manually revoke the object URL to avoid memory leaks.
+    if (textFile !== null) {
+      window.URL.revokeObjectURL(textFile);
+    }
+    textFile = window.URL.createObjectURL(data);
+    return textFile;
+  };
+
+  // Exports app data to a file
+  function exportFunc() {
+    // Get the storage element
+    chrome.storage.sync.get(null, function (pages) {
+      let exportText = JSON.stringify(pages);
+
+      // Add storage element to file
+      let link = document.createElement('a');
+      link.setAttribute('download', 'readinglist.json');
+      link.href = makeTextFile(exportText);
+      document.body.appendChild(link);
+
+      // wait for the link to be added to the document
+      window.requestAnimationFrame(function () {
+        var event = new MouseEvent('click');
+        link.dispatchEvent(event);
+        document.body.removeChild(link);
+      });
+    });
+  }
+
+  // Helper for import function
+  function impLoad() {
+    let myImportedData = JSON.parse(this.result);
+    //here is your imported data, and from here you should know what to do with it (save it to some storage, etc.)
+    chrome.storage.sync.clear();
```

Comment:
Getting rid of this line will preserve the existing data—but duplicates will still be overwritten (particularly the settings).